Tools/Pipeline/Gtk/MainWindow.cs

```diff
@@ -742,11 +742,26 @@ public void OnRenameActionActivated (object sender, EventArgs e)
             UpdateMenus();
             expand = false;
         }
-        
+
         public void OnDeleteActionActivated (object sender, EventArgs e)
         {
-            projectview1.Remove ();
-            UpdateMenus();
+            var dialog = new MessageDialog(this, DialogFlags.Modal, MessageType.Warning, ButtonsType.None, "How do you want to remove selected file/s?");
+            dialog.Title = "Delete File/s";
+            dialog.SecondaryText = "Do note that selecting \"Delete\" option will delete the file and this action will not be undoable.";
+
+            dialog.AddButton("Delete", ResponseType.Apply);
+            dialog.AddButton("Cancel", ResponseType.Cancel);
+            dialog.AddButton("Remove", ResponseType.Ok);
```

Comment:
MonoDevelop/Xamarin Studio has this type of delete dialog:

![deletedilalog](https://cloud.githubusercontent.com/assets/6773302/12548489/f4a3b5a6-c356-11e5-979b-ef4bdf13434c.png)